modules/parser/src/parsers/parse-stream-data-message.js

```diff
@@ -42,22 +46,71 @@ function isJSONString(str) {
   return (firstChar === '{' && lastChar === '}') || (firstChar === '[' && lastChar === ']');
 }
 
-const XVIZ_TYPE_PATTERN = /"type":\s*"xviz\//;
+const XVIZ_TYPE_PATTERN = /"type":\s*"(xviz\/\w*)"/;
+const XVIZ_TYPE_VALUE_PATTERN = /xviz\/\w*/;
 
-// returns true if the input represents an enveloped XVIZ object as a JSON string.
-// Can be either string or Uint8Array
-function isXVIZJSONString(str) {
-  // {"type":"xviz/
-  let firstChunk = str.slice(0, 14);
+// Returns the XVIZ message 'type' from the input strings
+// else null if not found.
+function getXVIZType(firstChunk, lastChunk) {
+  let result = firstChunk.match(XVIZ_TYPE_PATTERN);
+  if (!result && lastChunk) {
+    result = lastChunk.match(XVIZ_TYPE_PATTERN);
+  }
+
+  if (result) {
+    // return the first match group which contains the type
+    return result[1];
+  }
+
+  return null;
+}
+
+// Returns the XVIZ message 'type' from the input string
+// else null if not found.
+function getObjectXVIZType(type) {
+  const match = type.match(XVIZ_TYPE_VALUE_PATTERN);
+  if (match) {
+    return match[0];
+  }
+
+  return null;
+}
+
+// return the XVIZ type string if the input represents an enveloped XVIZ
+// object as a JSON string.
+// 'str' can be either string or Uint8Array
+function getJSONXVIZType(str) {
+  // We are trying to capture
+  // "type"\s*:\s*"xviz/transform_point_in_time"
+  // which the smallest is 37 bytes. Grab 50
+  // to provide room for spacing
+
+  // {"type":"xviz/*"
+  let firstChunk = str.slice(0, 50);
   // "type":"xviz/*"}
-  let lastChunk = str.slice(-36);
+  let lastChunk = str.slice(-50);
 
   if (Number.isFinite(firstChunk[0])) {
     firstChunk = String.fromCharCode.apply(null, firstChunk);
     lastChunk = String.fromCharCode.apply(null, lastChunk);
   }
 
-  return XVIZ_TYPE_PATTERN.test(firstChunk) || XVIZ_TYPE_PATTERN.test(lastChunk);
+  return getXVIZType(firstChunk, lastChunk);
+}
+
+function getBinaryXVIZType(arraybuffer) {
+  const jsonBuffer = getBinaryXVIZJSONBuffer(arraybuffer);
+  if (!jsonBuffer) {
+    return null;
+  }
+
+  // We have no choice but to decode the JSON portion of the buffer
+  // since it also contains all the GLB headers. This means we do not
+  // have any meaningful limits for where to search for the 'type' string
+  const textDecoder = new TextDecoder('utf8');
+  const jsonString = textDecoder.decode(jsonBuffer);
+
+  return getXVIZType(jsonString);
 }
 
 function getDataType(data) {
```

Comment:
Could you export `getDataType` at the module level? This is a useful method to use in conjunction with `getXVIZMessageType`